### PR TITLE
Add keepalive settings for adsc client

### PR DESCRIPTION
**Pull request by SpecialYang**
Using keepalive settings to detect the broken tcp connection sooner.

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
I think we have some standard one set throughout Istio, maybe we can reuse those?

**Comment by zirain** [comment]
> I think we have some standard one set throughout Istio, maybe we can reuse those?

`pkg/keepalive` ?

**Comment by SpecialYang** [comment]
> I think we have some standard one set throughout Istio, maybe we can reuse those?

Have updated. @howardjohn PTAL.

File: pilot/pkg/bootstrap/configcontroller.go [hunk 1]
```diff
@@ -251,6 +255,17 @@ func (s *Server) initConfigSources(args *PilotArgs) (err error) {
 					IstioRevision: args.Revision,
 				}.ToStruct(),
 				InitialDiscoveryRequests: adsc.ConfigInitialRequests(),
+				GrpcOpts: []grpc.DialOption{
+					grpc.WithKeepaliveParams(keepalive.ClientParameters{
```
Comment (howardjohn) [inline pilot/pkg/bootstrap/configcontroller.go:259]: This is missing MaxConnectionAge. Can we make a function that converts `pkg/keepalive.Options` to gRPC option so it is used consistent?

Comment (SpecialYang) [reply pilot/pkg/bootstrap/configcontroller.go]: The MaxConnectionAge is used for grpc server. Here we configure the grpc client,  so we only care about the keepalive interval and timeout.
Add a function for pkg/keepalive.Options to convert it to grpc client option used for grpc client keepalive.

**Comment by SpecialYang** [comment]
@howardjohn Any suggestion? Thanks.

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**